### DISCUSSION TITLE
New multiplication operators

### DISCUSSION THREAD

**Discussion by eeegs** [dotnet/csharplang#5052] [Language Ideas] [answered]
Hi,

I've recently been looking into Geometric Algebra.  It would be nice if C# have better support for some extra multiplication operators.

Could we add '⋅', '⨯', '⋀' & '⋁' as binary operators with the same precedence as multiplication, that initially only do multiplication.

This would have no real impact on any existing code but make for some way cool overloads.

Of cause if this could be extended to allow the introduction of arbitrary operators for overloading, that could be both uni and/or binary, then even better.

Regards,

Scott

**Comment by svick** [comment]
I think the big problem with this is that most people wouldn't know how to type such operators. That also means that the target audience for this feature would be likely very small and so the work required to implement this would likely not be worth it.

**Reply by eeegs** [reply]
Agreed, but that is why I am only asking for more operators that simply repeat the current functionality.  Surely that can't be too hard (famous last words).

**Comment by khm1600** [comment, marked as answer]
Related: https://github.com/dotnet/csharplang/discussions/175

**Reply by HaloFour** [reply]
Dang it, I was literally just typing that 😁 

IMO, adding four new operators that do multiplication doesn't sound remotely worthwhile, especially if the goal is only to allow overloading them to do something else.

**Comment by eeegs** [comment]
Ok, so no then ;)

ta
